### ISSUE TITLE
Compress::Zlib location conflicts with autosplit

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#10282]
Migrated from [rt.perl.org#74088](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=74088) (status was 'resolved')

Searchable as RT74088$

**Comment by p5pRT** [comment]
### From mls@suse.de

#### Created by mls@suse.de



The following code \(stripped down from Mail\-SpamAssassin\-3\.3\.1\) prints
an error under 5\.12\.0&#8203;:

&nbsp;    $SIG\{'\_\_DIE\_\_'\} = sub \{ print\("error @&#8203;\_"\) unless $^S; \};
&nbsp;    require Compress&#8203;::Zlib;

The difference between 5\.10\.1 and 5\.12\.0 is a change in the location of
Compress&#8203;::Zlib\. It used to live in archlib\, but now is in privlib\. The
autosplit files like autosplit\.ix get installed into archlib nevertheless\.

\(It's also weird that $^S is not set in the \_\_DIE\_\_ handler\, but 5\.10\.1
also shows this behaviour\.\)

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

This perlbug was built using Perl 5.12.0 - Tue Apr  6 13:36:06 UTC 2010
It is being executed now by  Perl 5.12.0 - Tue Apr  6 13:29:13 UTC 2010.

Site configuration information for perl 5.12.0:

Configured by abuild at Tue Apr  6 13:29:13 UTC 2010.

Summary of my perl5 (revision 5 version 12 subversion 0) configuration:
   
  Platform:
    osname=linux, osvers=2.6.32, archname=i586-linux-thread-multi
    uname='linux build33 2.6.32 #1 smp 2010-01-06 16:07:25 +0100 i686 athlon i386 gnulinux '
    config_args='-ds -e -Dprefix=/usr -Dvendorprefix=/usr -Dinstallusrbinperl -Dusethreads -Di_db -Di_dbm -Di_ndbm -Di_gdbm -Duseshrplib=true -Doptimize=-fomit-frame-pointer -fmessage-length=0 -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector -funwind-tables -fasynchronous-unwind-tables -Wall -pipe -Accflags=-DPERL_USE_SAFE_PUTENV'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=define, usemultiplicity=define
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=undef, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -DPERL_USE_SAFE_PUTENV -fno-strict-aliasing -pipe -fstack-protector -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-fomit-frame-pointer -fmessage-length=0 -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector -funwind-tables -fasynchronous-unwind-tables -Wall -pipe',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -DPERL_USE_SAFE_PUTENV -fno-strict-aliasing -pipe -fstack-protector'
    ccversion='', gccversion='4.5.0 20100331 (experimental) [trunk revision 157870]', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -fstack-protector'
    libpth=/lib /usr/lib /usr/local/lib
    libs=-lnsl -ldl -lm -lcrypt -lutil -lpthread -lc
    perllibs=-lnsl -ldl -lm -lcrypt -lutil -lpthread -lc
    libc=/lib/libc-2.11.1.so, so=so, useshrplib=true, libperl=libperl.so
    gnulibc_version='2.11.1'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E -Wl,-rpath,/usr/lib/perl5/5.12.0/i586-linux-thread-multi/CORE'
    cccdlflags='-fPIC', lddlflags='-shared -fomit-frame-pointer -fmessage-length=0 -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector -funwind-tables -fasynchronous-unwind-tables -Wall -pipe'

Locally applied patches:
    RC1


@INC for perl 5.12.0:
    /usr/lib/perl5/site_perl/5.12.0/i586-linux-thread-multi
    /usr/lib/perl5/site_perl/5.12.0
    /usr/lib/perl5/vendor_perl/5.12.0/i586-linux-thread-multi
    /usr/lib/perl5/vendor_perl/5.12.0
    /usr/lib/perl5/5.12.0/i586-linux-thread-multi
    /usr/lib/perl5/5.12.0
    .


Environment for perl 5.12.0:
    HOME=/root
    LANG (unset)
    LANGUAGE (unset)
    LC_COLLATE=POSIX
    LC_CTYPE=de_DE@euro
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/suse/mls/bin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/usr/X11R6/bin:/bin:/usr/lib/java/bin:/usr/games/bin:/usr/games:/opt/gnome/bin:/opt/kde/bin:/usr/openwin/bin:/opt/pilotsdk/bin:/suse/mls/korn
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @pmqs

Is the change below\, that was made a while back when Compress&#8203;::Zlib lost it's xs file\, responsible for the issue?

&nbsp;    \(
&nbsp;    $\] >= 5\.009 && $\] \<= 5\.011001 && \! $ENV\{PERL\_CORE\}
&nbsp;        ? \(INSTALLPRIVLIB    => '$\(INSTALLARCHLIB\)'\)
&nbsp;        : \(\)
&nbsp;    \)\,

If so\, does it need to change?

Paul

> \-\-\-\-\-Original Message\-\-\-\-\-
> From&#8203;: via RT \[mailto&#8203;:mls@&#8203;suse\.deperlbug\-followup@&#8203;perl\.org\]
> Sent&#8203;: 06 April 2010 17&#8203;:33
> To&#8203;: bugs\-bitbucket@&#8203;rt\.perl\.org
> Subject&#8203;: \[perl \#74088\] Compress&#8203;::Zlib location conflicts with autosplit
> 
> \# New Ticket Created by  mls@&#8203;suse\.de
> \# Please include the string&#8203;:  \[perl \#74088\]
> \# in the subject line of all future correspondence about this issue\.
> \# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt3/Ticket/Display\.html?id=74088 >
> 
> 
> This is a bug report for perl from mls@&#8203;suse\.de\,
> generated with the help of perlbug 1\.39 running under perl 5\.12\.0\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please describe your issue here\]
> 
> The following code \(stripped down from Mail\-SpamAssassin\-3\.3\.1\) prints
> an error under 5\.12\.0&#8203;:
> 
>     $SIG\{'\_\_DIE\_\_'\} = sub \{ print\("error @&#8203;\_"\) unless $^S; \};
>     require Compress&#8203;::Zlib;
> 
> The difference between 5\.10\.1 and 5\.12\.0 is a change in the location of
> Compress&#8203;::Zlib\. It used to live in archlib\, but now is in privlib\. The
> autosplit files like autosplit\.ix get installed into archlib
> nevertheless\.
> 
> \(It's also weird that $^S is not set in the \_\_DIE\_\_ handler\, but 5\.10\.1
> also shows this behaviour\.\)
> 
> 
> \[Please do not change anything below this line\]
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \-\-\-
> Flags&#8203;:
>     category=core
>     severity=medium
> \-\-\-
> This perlbug was built using Perl 5\.12\.0 \- Tue Apr  6 13&#8203;:36&#8203;:06 UTC 2010
> It is being executed now by  Perl 5\.12\.0 \- Tue Apr  6 13&#8203;:29&#8203;:13 UTC
> 2010\.
> 
> Site configuration information for perl 5\.12\.0&#8203;:
> 
> Configured by abuild at Tue Apr  6 13&#8203;:29&#8203;:13 UTC 2010\.
> 
> Summary of my perl5 \(revision 5 version 12 subversion 0\) configuration&#8203;:
> 
>   Platform&#8203;:
>     osname=linux\, osvers=2\.6\.32\, archname=i586\-linux\-thread\-multi
>     uname='linux build33 2\.6\.32 \#1 smp 2010\-01\-06 16&#8203;:07&#8203;:25 \+0100 i686
> athlon i386 gnulinux '
>     config\_args='\-ds \-e \-Dprefix=/usr \-Dvendorprefix=/usr \-
> Dinstallusrbinperl \-Dusethreads \-Di\_db \-Di\_dbm \-Di\_ndbm \-Di\_gdbm \-
> Duseshrplib=true \-Doptimize=\-fomit\-frame\-pointer \-fmessage\-length=0 \-O2
> \-Wall \-D\_FORTIFY\_SOURCE=2 \-fstack\-protector \-funwind\-tables \-
> fasynchronous\-unwind\-tables \-Wall \-pipe \-Accflags=\-
> DPERL\_USE\_SAFE\_PUTENV'
>     hint=recommended\, useposix=true\, d\_sigaction=define
>     useithreads=define\, usemultiplicity=define
>     useperlio=define\, d\_sfio=undef\, uselargefiles=define\,
> usesocks=undef
>     use64bitint=undef\, use64bitall=undef\, uselongdouble=undef
>     usemymalloc=n\, bincompat5005=undef
>   Compiler&#8203;:
>     cc='cc'\, ccflags ='\-D\_REENTRANT \-D\_GNU\_SOURCE \-
> DPERL\_USE\_SAFE\_PUTENV \-fno\-strict\-aliasing \-pipe \-fstack\-protector \-
> D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64'\,
>     optimize='\-fomit\-frame\-pointer \-fmessage\-length=0 \-O2 \-Wall \-
> D\_FORTIFY\_SOURCE=2 \-fstack\-protector \-funwind\-tables \-fasynchronous\-
> unwind\-tables \-Wall \-pipe'\,
>     cppflags='\-D\_REENTRANT \-D\_GNU\_SOURCE \-DPERL\_USE\_SAFE\_PUTENV \-fno\-
> strict\-aliasing \-pipe \-fstack\-protector'
>     ccversion=''\, gccversion='4\.5\.0 20100331 \(experimental\) \[trunk
> revision 157870\]'\, gccosandvers=''
>     intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234
>     d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=12
>     ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\,
> lseeksize=8
>     alignbytes=4\, prototype=define
>   Linker and Libraries&#8203;:
>     ld='cc'\, ldflags =' \-fstack\-protector'
>     libpth=/lib /usr/lib /usr/local/lib
>     libs=\-lnsl \-ldl \-lm \-lcrypt \-lutil \-lpthread \-lc
>     perllibs=\-lnsl \-ldl \-lm \-lcrypt \-lutil \-lpthread \-lc
>     libc=/lib/libc\-2\.11\.1\.so\, so=so\, useshrplib=true\,
> libperl=libperl\.so
>     gnulibc\_version='2\.11\.1'
>   Dynamic Linking&#8203;:
>     dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-Wl\,\-E \-
> Wl\,\-rpath\,/usr/lib/perl5/5\.12\.0/i586\-linux\-thread\-multi/CORE'
>     cccdlflags='\-fPIC'\, lddlflags='\-shared \-fomit\-frame\-pointer \-
> fmessage\-length=0 \-O2 \-Wall \-D\_FORTIFY\_SOURCE=2 \-fstack\-protector \-
> funwind\-tables \-fasynchronous\-unwind\-tables \-Wall \-pipe'
> 
> Locally applied patches&#8203;:
>     RC1
> 
> \-\-\-
> @&#8203;INC for perl 5\.12\.0&#8203;:
>     /usr/lib/perl5/site\_perl/5\.12\.0/i586\-linux\-thread\-multi
>     /usr/lib/perl5/site\_perl/5\.12\.0
>     /usr/lib/perl5/vendor\_perl/5\.12\.0/i586\-linux\-thread\-multi
>     /usr/lib/perl5/vendor\_perl/5\.12\.0
>     /usr/lib/perl5/5\.12\.0/i586\-linux\-thread\-multi
>     /usr/lib/perl5/5\.12\.0
>     \.
> 
> \-\-\-
> Environment for perl 5\.12\.0&#8203;:
>     HOME=/root
>     LANG \(unset\)
>     LANGUAGE \(unset\)
>     LC\_COLLATE=POSIX
>     LC\_CTYPE=de\_DE@&#8203;euro
>     LD\_LIBRARY\_PATH \(unset\)
>     LOGDIR \(unset\)
> 
> PATH=/suse/mls/bin&#8203;:/usr/local/bin&#8203;:/usr/bin&#8203;:/usr/sbin&#8203;:/sbin&#8203;:/usr/X11R6/b
> in&#8203;:/bin&#8203;:/usr/lib/java/bin&#8203;:/usr/games/bin&#8203;:/usr/games&#8203;:/opt/gnome/bin&#8203;:/opt
> /kde/bin&#8203;:/usr/openwin/bin&#8203;:/opt/pilotsdk/bin&#8203;:/suse/mls/korn
>     PERL\_BADLANG \(unset\)
>     SHELL=/bin/bash

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From mls@suse.de


A simple fix would be to add 'Compress/Zlib' to the archpms
hash in installperl\.

M\.

**Comment by p5pRT** [comment]
### From @pmqs

I've found the problem\. 

Compress&#8203;::Zlib still has an autoload section that was needed when it was an XS module\. I forgot to remove it when the XS code got moved to Compress&#8203;::Raw&#8203;::Zlib\. 

I'll sort out the CPAN copy\.

Paul

> \-\-\-\-\-Original Message\-\-\-\-\-
> From&#8203;: via RT \[mailto&#8203;:mls@&#8203;suse\.deperlbug\-followup@&#8203;perl\.org\]
> Sent&#8203;: 06 April 2010 17&#8203;:33
> To&#8203;: bugs\-bitbucket@&#8203;rt\.perl\.org
> Subject&#8203;: \[perl \#74088\] Compress&#8203;::Zlib location conflicts with autosplit
> 
> \# New Ticket Created by  mls@&#8203;suse\.de
> \# Please include the string&#8203;:  \[perl \#74088\]
> \# in the subject line of all future correspondence about this issue\.
> \# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt3/Ticket/Display\.html?id=74088 >
> 
> 
> This is a bug report for perl from mls@&#8203;suse\.de\,
> generated with the help of perlbug 1\.39 running under perl 5\.12\.0\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please describe your issue here\]
> 
> The following code \(stripped down from Mail\-SpamAssassin\-3\.3\.1\) prints
> an error under 5\.12\.0&#8203;:
> 
>     $SIG\{'\_\_DIE\_\_'\} = sub \{ print\("error @&#8203;\_"\) unless $^S; \};
>     require Compress&#8203;::Zlib;
> 
> The difference between 5\.10\.1 and 5\.12\.0 is a change in the location of
> Compress&#8203;::Zlib\. It used to live in archlib\, but now is in privlib\. The
> autosplit files like autosplit\.ix get installed into archlib
> nevertheless\.
> 
> \(It's also weird that $^S is not set in the \_\_DIE\_\_ handler\, but 5\.10\.1
> also shows this behaviour\.\)
> 
> 
> \[Please do not change anything below this line\]
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \-\-\-
> Flags&#8203;:
>     category=core
>     severity=medium
> \-\-\-
> This perlbug was built using Perl 5\.12\.0 \- Tue Apr  6 13&#8203;:36&#8203;:06 UTC 2010
> It is being executed now by  Perl 5\.12\.0 \- Tue Apr  6 13&#8203;:29&#8203;:13 UTC
> 2010\.
> 
> Site configuration information for perl 5\.12\.0&#8203;:
> 
> Configured by abuild at Tue Apr  6 13&#8203;:29&#8203;:13 UTC 2010\.
> 
> Summary of my perl5 \(revision 5 version 12 subversion 0\) configuration&#8203;:
> 
>   Platform&#8203;:
>     osname=linux\, osvers=2\.6\.32\, archname=i586\-linux\-thread\-multi
>     uname='linux build33 2\.6\.32 \#1 smp 2010\-01\-06 16&#8203;:07&#8203;:25 \+0100 i686
> athlon i386 gnulinux '
>     config\_args='\-ds \-e \-Dprefix=/usr \-Dvendorprefix=/usr \-
> Dinstallusrbinperl \-Dusethreads \-Di\_db \-Di\_dbm \-Di\_ndbm \-Di\_gdbm \-
> Duseshrplib=true \-Doptimize=\-fomit\-frame\-pointer \-fmessage\-length=0 \-O2
> \-Wall \-D\_FORTIFY\_SOURCE=2 \-fstack\-protector \-funwind\-tables \-
> fasynchronous\-unwind\-tables \-Wall \-pipe \-Accflags=\-
> DPERL\_USE\_SAFE\_PUTENV'
>     hint=recommended\, useposix=true\, d\_sigaction=define
>     useithreads=define\, usemultiplicity=define
>     useperlio=define\, d\_sfio=undef\, uselargefiles=define\,
> usesocks=undef
>     use64bitint=undef\, use64bitall=undef\, uselongdouble=undef
>     usemymalloc=n\, bincompat5005=undef
>   Compiler&#8203;:
>     cc='cc'\, ccflags ='\-D\_REENTRANT \-D\_GNU\_SOURCE \-
> DPERL\_USE\_SAFE\_PUTENV \-fno\-strict\-aliasing \-pipe \-fstack\-protector \-
> D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64'\,
>     optimize='\-fomit\-frame\-pointer \-fmessage\-length=0 \-O2 \-Wall \-
> D\_FORTIFY\_SOURCE=2 \-fstack\-protector \-funwind\-tables \-fasynchronous\-
> unwind\-tables \-Wall \-pipe'\,
>     cppflags='\-D\_REENTRANT \-D\_GNU\_SOURCE \-DPERL\_USE\_SAFE\_PUTENV \-fno\-
> strict\-aliasing \-pipe \-fstack\-protector'
>     ccversion=''\, gccversion='4\.5\.0 20100331 \(experimental\) \[trunk
> revision 157870\]'\, gccosandvers=''
>     intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234
>     d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=12
>     ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\,
> lseeksize=8
>     alignbytes=4\, prototype=define
>   Linker and Libraries&#8203;:
>     ld='cc'\, ldflags =' \-fstack\-protector'
>     libpth=/lib /usr/lib /usr/local/lib
>     libs=\-lnsl \-ldl \-lm \-lcrypt \-lutil \-lpthread \-lc
>     perllibs=\-lnsl \-ldl \-lm \-lcrypt \-lutil \-lpthread \-lc
>     libc=/lib/libc\-2\.11\.1\.so\, so=so\, useshrplib=true\,
> libperl=libperl\.so
>     gnulibc\_version='2\.11\.1'
>   Dynamic Linking&#8203;:
>     dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-Wl\,\-E \-
> Wl\,\-rpath\,/usr/lib/perl5/5\.12\.0/i586\-linux\-thread\-multi/CORE'
>     cccdlflags='\-fPIC'\, lddlflags='\-shared \-fomit\-frame\-pointer \-
> fmessage\-length=0 \-O2 \-Wall \-D\_FORTIFY\_SOURCE=2 \-fstack\-protector \-
> funwind\-tables \-fasynchronous\-unwind\-tables \-Wall \-pipe'
> 
> Locally applied patches&#8203;:
>     RC1
> 
> \-\-\-
> @&#8203;INC for perl 5\.12\.0&#8203;:
>     /usr/lib/perl5/site\_perl/5\.12\.0/i586\-linux\-thread\-multi
>     /usr/lib/perl5/site\_perl/5\.12\.0
>     /usr/lib/perl5/vendor\_perl/5\.12\.0/i586\-linux\-thread\-multi
>     /usr/lib/perl5/vendor\_perl/5\.12\.0
>     /usr/lib/perl5/5\.12\.0/i586\-linux\-thread\-multi
>     /usr/lib/perl5/5\.12\.0
>     \.
> 
> \-\-\-
> Environment for perl 5\.12\.0&#8203;:
>     HOME=/root
>     LANG \(unset\)
>     LANGUAGE \(unset\)
>     LC\_COLLATE=POSIX
>     LC\_CTYPE=de\_DE@&#8203;euro
>     LD\_LIBRARY\_PATH \(unset\)
>     LOGDIR \(unset\)
> 
> PATH=/suse/mls/bin&#8203;:/usr/local/bin&#8203;:/usr/bin&#8203;:/usr/sbin&#8203;:/sbin&#8203;:/usr/X11R6/b
> in&#8203;:/bin&#8203;:/usr/lib/java/bin&#8203;:/usr/games/bin&#8203;:/usr/games&#8203;:/opt/gnome/bin&#8203;:/opt
> /kde/bin&#8203;:/usr/openwin/bin&#8203;:/opt/pilotsdk/bin&#8203;:/suse/mls/korn
>     PERL\_BADLANG \(unset\)
>     SHELL=/bin/bash

**Comment by p5pRT** [comment]
### From @pmqs

From&#8203;: Michael Schroeder \[mailto&#8203;:mls@&#8203;suse\.de\]
 
> 
> A simple fix would be to add 'Compress/Zlib' to the archpms
> hash in installperl\.

Sorry\, no \-  Compress&#8203;::Zlib isn't arch dependent anymore\. 

Paul

**Comment by p5pRT** [comment]
### From @pmqs

From&#8203;: Michael Schroeder \[mailto&#8203;:mls@&#8203;suse\.de\]
 
> On Fri\, Apr 09\, 2010 at 05&#8203;:38&#8203;:07PM \+0100\, Paul Marquess wrote&#8203;:
> > From&#8203;: Michael Schroeder \[mailto&#8203;:mls@&#8203;suse\.de\]
> >
> > >
> > > A simple fix would be to add 'Compress/Zlib' to the archpms
> > > hash in installperl\.
> >
> > Sorry\, no \-  Compress&#8203;::Zlib isn't arch dependent anymore\.
> 
> Then please fix installperl so that Compress&#8203;::Zlib's autosplit\.ix
> file doesn't get installed into perlarch\. Currently is assumes
> everything in the 'auto' directory is arch dependent\.

Yep\, I have that sorted\.

Paul

**Comment by p5pRT** [comment]
### From mls@suse.de

On Fri\, Apr 09\, 2010 at 05&#8203;:38&#8203;:07PM \+0100\, Paul Marquess wrote&#8203;:
> From&#8203;: Michael Schroeder \[mailto&#8203;:mls@&#8203;suse\.de\]
>  
> > 
> > A simple fix would be to add 'Compress/Zlib' to the archpms
> > hash in installperl\.
> 
> Sorry\, no \-  Compress&#8203;::Zlib isn't arch dependent anymore\. 

Then please fix installperl so that Compress&#8203;::Zlib's autosplit\.ix
file doesn't get installed into perlarch\. Currently is assumes
everything in the 'auto' directory is arch dependent\.

Cheers\,
&nbsp;  Michael\.

\-\- 
Michael Schroeder                                   mls@&#8203;suse\.de
SUSE LINUX Products GmbH\, GF Markus Rex\, HRB 16746 AG Nuernberg
main\(\_\)\{while\(\_=~getchar\(\)\)putchar\(~\_\-1/\(~\(\_|32\)/13\*2\-11\)\*13\);\}

**Comment by p5pRT** [comment]
### From @iabyn

On Fri\, Apr 09\, 2010 at 11&#8203;:48&#8203;:05PM \+0100\, Paul Marquess wrote&#8203;:
> From&#8203;: Michael Schroeder \[mailto&#8203;:mls@&#8203;suse\.de\]
>  
> > On Fri\, Apr 09\, 2010 at 05&#8203;:38&#8203;:07PM \+0100\, Paul Marquess wrote&#8203;:
> > > From&#8203;: Michael Schroeder \[mailto&#8203;:mls@&#8203;suse\.de\]
> > >
> > > >
> > > > A simple fix would be to add 'Compress/Zlib' to the archpms
> > > > hash in installperl\.
> > >
> > > Sorry\, no \-  Compress&#8203;::Zlib isn't arch dependent anymore\.
> > 
> > Then please fix installperl so that Compress&#8203;::Zlib's autosplit\.ix
> > file doesn't get installed into perlarch\. Currently is assumes
> > everything in the 'auto' directory is arch dependent\.
> 
> Yep\, I have that sorted\.

Is this something that will need pulling in for 5\.12\.1?

\-\- 
Lady Nancy Astor&#8203;: If you were my husband\, I would flavour your coffee
with poison\.
Churchill&#8203;: Madam \- if I were your husband\, I would drink it\.

**Comment by p5pRT** [comment]
### From @pmqs



> \-\-\-\-\-Original Message\-\-\-\-\-
> From&#8203;: Dave Mitchell \[mailto&#8203;:davem@&#8203;iabyn\.com\]
> Sent&#8203;: 12 April 2010 17&#8203;:18
> To&#8203;: Paul Marquess
> Cc&#8203;: 'Michael Schroeder'; perl5\-porters@&#8203;perl\.org; perlbug\-
> followup@&#8203;perl\.org
> Subject&#8203;: Re&#8203;: \[perl \#74088\] perlbug AutoReply&#8203;: Compress&#8203;::Zlib location
> conflicts with autosplit
> 
> On Fri\, Apr 09\, 2010 at 11&#8203;:48&#8203;:05PM \+0100\, Paul Marquess wrote&#8203;:
> > From&#8203;: Michael Schroeder \[mailto&#8203;:mls@&#8203;suse\.de\]
> >
> > > On Fri\, Apr 09\, 2010 at 05&#8203;:38&#8203;:07PM \+0100\, Paul Marquess wrote&#8203;:
> > > > From&#8203;: Michael Schroeder \[mailto&#8203;:mls@&#8203;suse\.de\]
> > > >
> > > > >
> > > > > A simple fix would be to add 'Compress/Zlib' to the archpms
> > > > > hash in installperl\.
> > > >
> > > > Sorry\, no \-  Compress&#8203;::Zlib isn't arch dependent anymore\.
> > >
> > > Then please fix installperl so that Compress&#8203;::Zlib's autosplit\.ix
> > > file doesn't get installed into perlarch\. Currently is assumes
> > > everything in the 'auto' directory is arch dependent\.
> >
> > Yep\, I have that sorted\.
> 
> Is this something that will need pulling in for 5\.12\.1?

I'd say\, yes\, it is\. 

That can be achieved by either syncing core with the CPAN copy of
Compress&#8203;::Zlib  \(when I actually upload it\)\, or just fixing this particular
issue\. I suspect the latter if we follow Jesse's rules\-of\-engagement for
5\.12\.X

Paul

**Comment by p5pRT** [comment]
### From @pmqs

Here is a patch that addresses \#74088 

I'll include this in the next CPAN update for Compress&#8203;::Zlib

Paul

**Comment by p5pRT** [comment]
### From @pmqs

<details><summary>0001-perl-74088.patch</summary>

```diff
From 571d8c96cf5d0683212052ceac7c8e53f40b3f3c Mon Sep 17 00:00:00 2001
From: paul <paul@paul-desktop.(none)>
Date: Thu, 15 Apr 2010 19:35:44 +0100
Subject: [PATCH] perl-74088

---
 cpan/IO-Compress/Changes              |    6 ++++++
 cpan/IO-Compress/lib/Compress/Zlib.pm |   15 ++-------------
 2 files changed, 8 insertions(+), 13 deletions(-)

diff --git a/cpan/IO-Compress/Changes b/cpan/IO-Compress/Changes
index c98bef1..e3a7a70 100644
--- a/cpan/IO-Compress/Changes
+++ b/cpan/IO-Compress/Changes
@@ -1,6 +1,12 @@
 CHANGES
 -------
 
+  2.024_01 15 April 2010
+
+      * Compress::Zlib
+        Remove autoload code from Zlib.pm.
+        [perl #74088]
+
   2.024 7 January 2010
 
       * Compress::Zlib
diff --git a/cpan/IO-Compress/lib/Compress/Zlib.pm b/cpan/IO-Compress/lib/Compress/Zlib.pm
index 9424df6..b546dac 100644
--- a/cpan/IO-Compress/lib/Compress/Zlib.pm
+++ b/cpan/IO-Compress/lib/Compress/Zlib.pm
@@ -3,7 +3,6 @@ package Compress::Zlib;
 
 require 5.004 ;
 require Exporter;
-use AutoLoader;
 use Carp ;
 use IO::Handle ;
 use Scalar::Util qw(dualvar);
@@ -16,9 +15,9 @@ use IO::Uncompress::Gunzip 2.024 ;
 use strict ;
 use warnings ;
 use bytes ;
-our ($VERSION, $XS_VERSION, @ISA, @EXPORT, @EXPORT_OK, %EXPORT_TAGS, $AUTOLOAD);
+our ($VERSION, $XS_VERSION, @ISA, @EXPORT, @EXPORT_OK, %EXPORT_TAGS);
 
-$VERSION = '2.024';
+$VERSION = '2.024_01';
 $XS_VERSION = $VERSION; 
 $VERSION = eval $VERSION;
 
@@ -46,16 +45,6 @@ BEGIN
     *zlib_version = \&Compress::Raw::Zlib::zlib_version;
 }
 
-sub AUTOLOAD {
-    my($constname);
-    ($constname = $AUTOLOAD) =~ s/.*:://;
-    my ($error, $val) = Compress::Raw::Zlib::constant($constname);
-    Carp::croak $error if $error;
-    no strict 'refs';
-    *{$AUTOLOAD} = sub { $val };
-    goto &{$AUTOLOAD};
-}
-
 use constant FLAG_APPEND             => 1 ;
 use constant FLAG_CRC                => 2 ;
 use constant FLAG_ADLER              => 4 ;
-- 
1.6.0.4


```
</details>

**Comment by p5pRT** [comment]
### From @cpansprout

This was fixed by commit e8796d6170\.

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'resolved'
